static/js/message_util.js

```diff
@@ -36,4 +36,24 @@ exports.get_messages_in_topic = function (stream_id, topic) {
     });
 };
 
+exports.delete_message = function (msg_id) {
+    const message = message_store.get(msg_id);
+    if (message === undefined) {
+        return;
+    }
```

Comment:
I think the `message_store` check is wrong -- we need to update `unread_ops` data, at least (which has a full picture from the server in all cases), even if we don't have the message's content cached in `message_store`.

Reply:
(Probably worth adding a comment explaining this detail in this function after fixing this)